cpp/ql/src/semmle/code/cpp/models/implementations/StdString.qll

```diff
@@ -12,3 +12,19 @@ class StdStringCStr extends TaintFunction {
     output.isReturnValue()
   }
 }
+
+/**
+ * The `std::string` function `append`.
+ */
+class StdStringAppend extends TaintFunction {
+  StdStringAppend() { this.hasQualifiedName("std", "basic_string", "append") }
+
+  override predicate hasTaintFlow(FunctionInput input, FunctionOutput output) {
+    // flow from parameter to string itself (qualifier) and return value
+    input.isParameterDeref(0) and
```

Comment:
There are [some overrides](https://en.cppreference.com/w/cpp/string/basic_string/append) with two or three parameters - have you looked through them and decided that taint only propagates from the first parameter in those cases?

Reply:
Should we also take `CharT`? I think that's a single character of the `basic_string`'s character type.

Reply:
I had thought no, since it can only take a small range of values.  But thinking about it a bit deeper, in some cases it might only take one or a few characters appended to create something malicious.  I think I'll add it...

Reply:
Done.  Added a test case for `CharT` as well, since I really didn't have much confidence without it.